warp10/src/main/java/io/warp10/script/filler/FillerNext.java

```diff
@@ -62,4 +65,31 @@ public int getPostWindow() {
   public int getPreWindow() {
     return 0;
   }
+
+  @Override
+  public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException {
+
+    // fill always sort input before.
+    final int nTicks = gts.size();
+    final GeoTimeSerie original = gts;
+    final long nextTick = GTSHelper.tickAtIndex(original, 0);
+    final long lastTick = GTSHelper.tickAtIndex(original, nTicks - 1);
+    final int[] currentIndex = {0};
```

Comment:
Why not make currentIndex a field of the instance you create below instead? This would make the code way easier to understand.